rwengine/src/data/GameTexts.hpp

```diff
@@ -3,21 +3,42 @@
 #include <string>
 #include <unordered_map>
 
+/**
+ * The game stores strings as 16-bit indexes into the font
+ * texture, which is something simllar to ASCII.
+ */
+using GameString = std::basic_string<char16_t>;
+/**
+ * Keys are small so should be subject to SSO
```

Comment:
Small string optimisation. Since keys are always <= 8 [most implementations](http://stackoverflow.com/questions/27631065/why-does-libcs-implementation-of-stdstring-take-up-3x-memory-as-libstdc/28003328#28003328) will always store the key string inside the string object. Not a big deal for performance but it means taking keys as rvalues references has no benefit.